templates/commands/goldentest/new_test_cli.go

```diff
@@ -0,0 +1,165 @@
+package goldentest
+
+// This file implements the "templates golden-test new-test" subcommand.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/abc/templates/common/input"
+	"github.com/abcxyz/abc/templates/common/specutil"
+	"github.com/abcxyz/abc/templates/common/templatesource"
+	"github.com/abcxyz/abc/templates/model/decode"
+	goldentest "github.com/abcxyz/abc/templates/model/goldentest/v1beta3"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+)
+
+type NewTestCommand struct {
+	cli.BaseCommand
+
+	flags NewTestFlags
+}
+
+func (c *NewTestCommand) Desc() string {
+	return "create a new golden test"
+}
+
+func (c *NewTestCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <test_name> <location>
+
+The {{ COMMAND }} create a new golden test.
+
+The "<test_name>" is the name of the test.
+
+The "<location>" is the location of the template.
+`
+}
+
+func (c *NewTestCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+	c.flags.Register(set)
+	return set
+}
+
+func (c *NewTestCommand) Run(ctx context.Context, args []string) (rErr error) {
+	logger := logging.FromContext(ctx)
+
+	if err := c.Flags().Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	tempDir, err := os.MkdirTemp("", "abc-new-test-*")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory: %w", err)
+	}
+	defer func() {
+		rErr = errors.Join(rErr, os.RemoveAll(tempDir))
+	}()
+
+	wd, err := c.WorkingDir()
+	if err != nil {
+		return fmt.Errorf("failed to get working directory: %w", err)
+	}
+
+	fs := &common.RealFS{}
+	logger.DebugContext(ctx, "downloading/copying template")
+	_, templateDir, err := templatesource.Download(ctx, &templatesource.DownloadParams{
+		CWD:         wd,
+		FS:          fs,
+		TempDirBase: tempDir,
+		Source:      c.flags.Location,
+		// using localSourceParser instead of remoteGitSourceParser.
+		GitProtocol: "",
+	})
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+
+	spec, err := specutil.Load(ctx, fs, templateDir, c.flags.Location)
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+	logger.DebugContext(ctx, "resolving inputs")
+
+	resolvedInputs, err := input.Resolve(ctx, &input.ResolveParams{
+		FS:       fs,
+		Inputs:   c.flags.Inputs,
+		Prompt:   c.flags.Prompt,
+		Prompter: c,
+		Spec:     spec,
+	})
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+
+	buf, err := marshalTestCase(resolvedInputs)
+	if err != nil {
+		return fmt.Errorf("failed to marhsal test config data: %w", err)
+	}
+
+	testDir := filepath.Join(c.flags.Location, goldenTestDir, c.flags.NewTestName)
+	testConfigFile := filepath.Join(testDir, configName)
+
+	err = fs.MkdirAll(testDir, common.OwnerRWXPerms)
+	if err != nil {
+		return fmt.Errorf("failed creating %s directory to contain test yaml file: %w", testDir, err)
+	}
+	// file overriding is not allowed.
```

Comment:
Maybe we can add forceOverwrite option to customize overwrite behavior?

Reply:
That sounds good to me.

Reply:
will do it in another PR